### PR TITLE
Add support for darwin_arm64

### LINKED ISSUE #21: Release new version with binary for Apple Silicon devices
Hi there,

one of our colleagues is working with a MacBook that uses an Apple Silicon CPU and it is always troublesome to compile the provider ourselves and google where you have to place the binary. It would be great if you could release a new version that includes a binary for the `darwin_arm64` platform!

### PR DESCRIPTION
**Pull request by yussufsh**
Fixes #21

Signed-off-by: Yussuf Shaikh <yussuf.shaikh1@ibm.com>

### CODE REVIEW

**Comment by yussufsh** [comment]
Verified running `goreleaser` locally.

```
% goreleaser release
  • starting release...
  • loading config file                              file=.goreleaser.yml
  • loading environment variables
  • getting and validating git state
    • building...                                    commit=1f8e46ffb118a4723915e53ccfc28854e137c65c latest tag=v2.1.3
  • parsing tag
  • setting defaults
  • running before hooks
    • running                                        hook=go mod tidy
  • checking distribution directory
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/terraform-provider-ignition_linux_386/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_linux_amd64_v1/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_freebsd_arm64/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_freebsd_arm_6/terraform-provi
Add support for darwin_arm64
der-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_windows_amd64_v1/terraform-provider-ignition_v2.1.3.exe
    • building                                       binary=dist/terraform-provider-ignition_freebsd_386/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_freebsd_amd64_v1/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_windows_386/terraform-provider-ignition_v2.1.3.exe
    • building                                       binary=dist/terraform-provider-ignition_linux_arm_6/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_linux_arm64/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_darwin_amd64_v1/terraform-provider-ignition_v2.1.3
    • building                                       binary=dist/terraform-provider-ignition_darwin_arm64/terraform-provider-ignition_v2.1.3
    • took: 4m33s
  • archives
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_freebsd_amd64.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_linux_arm.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_freebsd_arm64.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_linux_arm64.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_freebsd_arm.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_windows_amd64.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_windows_386.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_linux_386.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_darwin_amd64.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_darwin_arm64.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_freebsd_386.zip
    • creating                                       archive=dist/terraform-provider-ignition_2.1.3_linux_amd64.zip
    • took: 8s
  • calculating checksums
  • signing artifacts
    • refreshing checksums                           file=terraform-provider-ignition_2.1.3_SHA256SUMS
.
.
.
.

```

File: .goreleaser.yml [hunk 1]
```diff
@@ -29,6 +29,12 @@ builds:
   ignore:
     - goos: darwin
       goarch: '386'
+    - goos: darwin
+      goarch: 'arm'
```
Comment (bgilbert) [inline .goreleaser.yml]: Nit: `'386'` needs to be quoted but the other values don't.

**Review by bgilbert — APPROVED** [review]
LGTM

File: .goreleaser.yml (hunk 1 above)
Comment (yussufsh) [reply .goreleaser.yml]: done.

**Review by bgilbert — APPROVED** [review]
LGTM

**Pull request merged**